dds/idl/metaclass_generator.cpp

```diff
@@ -64,19 +65,15 @@ namespace {
     const std::string& firstArg, bool skip = false)
   {
     const size_t n = fieldName.size() + 1 /* 1 for the dot */;
-    const std::string fieldType = scoped(field->field_type()->name());
+    const std::string fieldType = Field::get_type_name(*(field->field_type()));
     be_global->impl_ <<
-      "    if (std::strncmp(field, \"" << fieldName << ".\", " << n
-      << ") == 0) {\n"
-      "      return getMetaStruct<" << fieldType << ">().getValue("
-      << firstArg << ", field + " << n << ");\n"
+      "    if (std::strncmp(field, \"" << fieldName << ".\", " << n << ") == 0) {\n"
```

Comment:
revert changes that only re-arrange the existing code (change how it's broken into different lines)